### PR TITLE
object-oriented [I2C_mutex] module

### PR DESCRIPTION
**Pull request by RobertLucian**
In this PR, the procedural functions inside the `I2C_mutex` module were replaced with a class that does the same thing + has procedures for cleaning the object.

Also, the `EasyGoPiGo.reset` method didn't work, so I had to fix it by making `DEFAULT_SPEED` a variable of the `EasyGoPiGo` class.

The new `I2C_mutex` has only been tested on the `EasyGoPiGo` class. 
Other classes such as `DigitalSensor` and `AnalogSensor` need to be tested aswell - just to be extremely sure they work.

Automated tests could have come in really handy in this situation.

### CODE REVIEW

File: Software/Python/easygopigo.py [hunk 1]
```diff
@@ -46,56 +46,33 @@ def debug(in_str):
...
+        self.DEFAULT_SPEED = 128
+        gopigo.set_speed(self.DEFAULT_SPEED)
+
+        self.mutex = None
+        if use_mutex is True:
+            self.mutex = Mutex()
+
     def volt(self):
-        _grab_read()
+        if self.mutex:
```
Comment (CleoQc) [inline Software/Python/easygopigo.py]: For code clarity, I really feel that this should be all done in one line, hence my use of _grab_read() i

Comment (CleoQc) [inline Software/Python/easygopigo.py]: How about having mutex.acquire() and mutex.release()  receive a Bool argument. If False, they just do nothing. At least easygopigo will be be burdened by twice the number of lines.  There are other approaches too that won't touch the mutex class, like creating a base method similar to _grab_read/release_read.

Comment (RobertLucian) [inline Software/Python/easygopigo.py]: I like your approach of sending a bool argument. I think we should go with this one.
Tomorrow (Friday), I'll do this thing.

**Comment by RobertLucian** [comment]
This is done @CleoQc .

**Comment by RobertLucian** [comment]
@CleoQc I think this done should get merged ASAP - the last time we looked at this one it was fine.

**Comment by RobertLucian** [comment]
@CleoQc these have been tested on a GoPiGo2 and the I2C mutex seems to be working fine.

**Pull request merged**